packages/data-context/src/actions/ProjectActions.ts

```diff
@@ -36,6 +36,7 @@ export interface ProjectApiShape {
   getCurrentBrowser: () => Cypress.Browser | undefined
   getCurrentProjectSavedState(): {} | undefined
   setPromptShown(slug: string): void
+  setSpecFilter(specFilter: string): void
```

Comment:
I am not sure we need new endpoints/code for this - can we just use the existing preferences code? It all ends up in the same object:

![image](https://user-images.githubusercontent.com/19196536/180117037-0ac39e38-3e2a-4171-a4ef-12bb33cad752.png)

For example here: https://github.com/cypress-io/cypress/blob/develop/packages/data-context/src/actions/LocalSettingsActions.ts#L13-L16 (I think you implemented the original preference code too). It seems like we are adding a specific endpoint for one preference, when we can consolidate it in one place. What do you think?

Reply:
At a glance I was expecting to do that - before picking this up we all thought this was just a trivial addition to this logic. But there are actually two different state objects and patterns: the existing `setPreference`/`localSettings` stuff manages values that apply to Cypress generally on the machine (stored in the `__global__/state.json`), and the one in your screenshot is in the project-specific `state.json`. 

We could refactor to have the same pattern for both kinds of settings, and just `setLocalPreference` and `setProjectPreference` though. I'd actually like that a lot but it seems like it would muddy this PR.

Reply:
It seems like this refactor would make a lot of sense though - if we don't do it now, I don't think it will ever happen, since it's very hard to justify going back and reworking code that already works great.

I'll leave this decision to you - I am not sure how much work it is, but a unified preferences API sure would be nice, instead of multiple code paths.

```js
setPreferences(json: any, type: 'global' | 'project') 
```

Or something to that meaning.

Reply:
After letting this sit for a bit, I started coming around as well. At a minimum, combining `setSpecFilter` and `setPromptShown` into `saveProjectPreferences' seems better than keeping them separated, since both work at the project level, and as Mike pointed out we can expect to maybe save other filtering criteria pretty soon. The only drawback is `setPromptShown` is a little more than just a simple save so we do have to shuffle things a little bit, maybe just created the object of the client instead of in `makeDataContext`.

What I really want to do is wrap this up and grab #21879, which together make a big improvement in the search experience. That pushes me towards what seems like the simpler solution, rather than fully combining both kinds of preferences... which, there's a bit to talk about there. Our names for things are a bit confusing and I'd be tempted to try to make it all make more sense.

If we come back with more things to save, we could combine them at that point. But really we don't seem to actually touch this code very much at the moment, values aren't added a lot.

Reply:
@mike-plummer @warrensplayer just wanted to mention I'll do this as part of the PR (should be ready to review tomorrow) and `saveProjectPreferences` should be useful for much of the ACI work, so might be worth chatting.

Reply:
Update, been exploring a bit around an _existing_ mutation called `setProjectPreferences`, which actually saves data in a third place, the `/Application Support/Cypress/cy/{environment}/cache` file. This file caches values for `lastBrowser` and `testingType` in objects that refer to each project by a simple string name, like `PROJECT_PREFERENCES['frontend-shared]` whereas the data in the projects folder is nested under more unique names with hashes (`frontend-shared-b1d9d76cfc89097c4beaf3412d05b7e0`), since two projects called `frontend-shared` could be in different directories and by opened by Cypress. So any cache based on project name alone is a little frail in my book.

The main purpose of that cache seems to be to maintain `USER` object with email address, auth token, and name, and the `PROJECTS` which is a list of global mode projects.

What this means:

- I'm not going to leverage that existing top-level cache, I'm going to continue to save data in the `state.json` under the `project+hash` directories where `firstOpened` etc are stored.
- Will rename `setProjectPreferences` to `setProjectPreferencesInGlobalCache` or something
- I'll make a chore ticket to move project-level saved settings out of that cache and keep it all together in the state.json (or have it explained why that's not the best solution)
- I will do Lachlan's suggestion of `setPreferences(json: any, type: 'global' | 'project')`, again to minimize confusion among all of these very related concepts.